testsuite/MDAnalysisTests/coordinates/test_chainreader.py

```diff
@@ -205,8 +205,9 @@ def test_set_one_format_tuple(self):
         assert universe.trajectory.n_frames == 99
 
     def test_set_all_formats(self):
-        universe = mda.Universe(PSF, [PDB_small, PDB_closed], format='pdb')
```

Comment:
ALso, see comments above re: `TypeError`

Reply:
The function looks like it wants to test whether `format='pdb'` sets the `ChainReader` to read all the frames of the trajectory as PDBs. It passes, but only because with the old code `ChainReader` would look at each file and find that it was a PDB. It also doesn't set the format of the topology.

In the flow of the original function + code:
- `topology_format = format = 'pdb'`
- `parser = get_parser_for(PSF, format='pdb')` -> `KeyError` -> `parser = get_reader_for(PSF)` -> PSF is parsed as PSF
- `format` not passed to `ChainReader` -> each frame uses `get_reader_for` to get a PDB reader

If the test is rewritten as `universe = mda.Universe(PSF, [PDB_small, PDB_closed], format='not_a_real_format')` it still passes.

It does sound like a good idea to rename the new test and add one that also reads in coordinates from different formats without raising an error, though.

Reply:
Oops, thanks for looking into this. I agree with your solution then.